src/iter/collect/mod.rs

```diff
@@ -91,32 +92,44 @@ impl<'c, T: Send + 'c> Collect<'c, T> {
         let start = self.vec.len();
         let mut slice = &mut self.vec[start..];
         slice = unsafe { slice::from_raw_parts_mut(slice.as_mut_ptr(), self.len) };
-        CollectConsumer::new(&self.writes, slice)
+        CollectConsumer::new(slice)
     }
 
     /// Updates the final vector length.
-    fn complete(self) {
+    fn complete(self, _result: VerifiedCollectResult) {
```

Comment:
Yes, the VerifiedCollectResult is not "tagged" so if you have two Collect, you can mix them up.